packages/client/src/v2-events/features/debug/debug.tsx

```diff
@@ -10,12 +10,46 @@
  */
 /* stylelint-disable */
 import { ReactQueryDevtools } from '@tanstack/react-query-devtools'
-import React from 'react'
+import { onlineManager } from '@tanstack/react-query'
+
+import React, { useEffect } from 'react'
+import styled from 'styled-components'
+
+const wasOffline = localStorage.getItem('debugger-offline') === 'true'
+
+onlineManager.setOnline(!wasOffline)
+
+/*
+ * React query Devtool assumes offline mode is never set from outside
+ * thus the button's state always first defaults to online.
+ * As we're setting the mode offline even before the component mounts,
+ * we need to override the default styles to indicate that the app is offline.
+ * This is a workaround for the issue.
+ */
+const Container = styled.div<{ isOnline: boolean }>`
```

Comment:
<img width="517" alt="image" src="https://github.com/user-attachments/assets/bef7c6e8-f6d2-4ea0-a823-aa0a43e2ba98" />
Workaround to make the Devtool show the offline mode correctly after page reload